diff --git a/meson.build b/meson.build
index xxxxxx..yyyyyy 100644
--- a/meson.build
+++ b/meson.build
@@ -1433,7 +1433,10 @@ if not get_option('opengl').auto() or have_system or have_vhost_user_gpu
   epoxy = dependency('epoxy', method: 'pkg-config',
                       required: get_option('opengl'))
   
+  # Make EGL headers optional when using OpenGL Core
+  need_egl = not get_option('opengl_core').enabled()
+  
-  if cc.has_header('epoxy/egl.h', dependencies: epoxy)
+  if (not need_egl) or cc.has_header('epoxy/egl.h', dependencies: epoxy)
     opengl = epoxy
   elif get_option('opengl').enabled()
     error('epoxy/egl.h not found')